xdsl/interpreter.py

```diff
@@ -265,9 +304,80 @@ def run(self, op: Operation):
         )
         self.set_values(zip(op.results, results))
 
+    def run_terminator(self, terminator: Operation):
+        """
+        Fetches the implemetation for the given terminator, passes it the Python values
+        associated with the SSA operands.
+        The Terminator implementation should return a Block | None, designating the target
```

Comment:
I would actually recommend adding infrastructure to interpreter functions to register an op as a terminator or not, with different function interfaces, etc. That way we can also catch errors early, like terminator ops registered as normal ops, and the first return value not being Block | None